geonode/layers/views.py

```diff
@@ -429,12 +432,119 @@ def layer_detail(request, layername, template='layers/layer_detail.html'):
     if settings.SOCIAL_ORIGINS:
         context_dict["social_links"] = build_social_links(request, layer)
 
+
```

Comment:
@karakostis I was talking about this whole chunk of code, is it related to your feature?

Reply:
Yes. This part of code is related to the feature.

Reply:
So the issue must be related with the that chunk of code. As I see in the log file something breaks there and returns the Exception message:

LL: <CDLL 'libc.so.6', handle 7fb2157ee9b0 at 7fb1f573ce90>
.on_geoserver_messages: RECEIVED MSG - body: {'rating': '0', 'constraints_other': 'None', 'bbox_x0': 'None', 'csw_mdsource': 'local', 'csw_schema': 'http://www.isotc211.org/2005/gmd', 'csw_typename': 'gmd:MD_Metadata', 'date_type': 'publication', 'srid': 'EPSG:4326', 'bbox_x1': 'None', 'edition': 'None', 'featured': 'False', 'owner': 'None', 'detail_url': 'None', 'id': 1, 'category': 'None', 'license': 'None', 'spatial_representation_type': 'None', 'uuid': u'', 'title': u'', 'alternate': 'None', 'regions': '[]', 'thumbnail_url': 'None', 'share_count': '0', 'abstract': u'', 'bbox_y0': 'None', 'bbox_y1': 'None', 'popular_count': '0', 'contacts': '[]', 'purpose': 'None', 'date': '2017-09-26 15:58:14.589434', 'group': 'None', 'temporal_extent_end': 'None', 'language': 'eng', 'data_quality_statement': 'None', 'restriction_code_type': 'None', 'maintenance_frequency': 'None', 'csw_anytext': 'None', 'metadata_xml': '<gmd:MD_Metadata xmlns:gmd="http://www.isotc211.org/2005/gmd"/>', 'tkeywords': '[]', 'metadata_uploaded_preserve': 'False', 'csw_type': 'dataset', 'keywords': '[]', 'metadata_uploaded': 'False', 'csw_wkt_geometry': 'POLYGON((-180 -90,-180 90,180 90,180 -90,-180 -90))', 'temporal_extent_start': 'None', 'is_published': True}
Layer matching query does not exist.
Traceback (most recent call last):
  File "/home/travis/build/GeoNode/geonode/geonode/geoserver/signals.py", line 94, in geoserver_post_save_local
    instance = Layer.objects.get(id=layer_id)
  File "/home/travis/virtualenv/python2.7_with_system_site_packages/local/lib/python2.7/site-packages/django/db/models/manager.py", line 127, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/home/travis/virtualenv/python2.7_with_system_site_packages/local/lib/python2.7/site-packages/django/db/models/query.py", line 334, in get
    self.model._meta.object_name
DoesNotExist: Layer matching query does not exist.
on_geoserver_messages: finished
finished.

Reply:
it is not clear, a geoserver post_save signal cannot find the resource for which it was called. Strange.